core/chaincode/platforms/util/writer.go

```diff
@@ -13,102 +13,67 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path"
 	"path/filepath"
 	"strings"
 	"time"
 
-	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/internal/ccmetadata"
 	"github.com/pkg/errors"
 )
 
-var vmLogger = flogging.MustGetLogger("container")
-
 // WriteFolderToTarPackage writes source files to a tarball.
 // This utility is used for node js chaincode packaging, but not golang chaincode.
 // Golang chaincode has more sophisticated file packaging, as implemented in golang/platform.go.
 func WriteFolderToTarPackage(tw *tar.Writer, srcPath string, excludeDirs []string, includeFileTypeMap map[string]bool, excludeFileTypeMap map[string]bool) error {
-	fileCount := 0
-	rootDirectory := srcPath
-
-	// trim trailing slash if it was passed
-	if rootDirectory[len(rootDirectory)-1] == '/' {
-		rootDirectory = rootDirectory[:len(rootDirectory)-1]
-	}
-
-	vmLogger.Debugf("rootDirectory = %s", rootDirectory)
+	rootDirectory := filepath.Clean(srcPath)
 
-	//append "/" if necessary
-	updatedExcludeDirs := make([]string, 0)
-	for _, excludeDir := range excludeDirs {
-		if excludeDir != "" && strings.LastIndex(excludeDir, "/") < len(excludeDir)-1 {
-			excludeDir = excludeDir + "/"
-			updatedExcludeDirs = append(updatedExcludeDirs, excludeDir)
-		}
```

Comment:
just making sure i understand this correctly. if an excluded dir does end with "/", it would not get appended to `updatedExcludeDirs`. This is wrong and fixed in this commit, right?

Reply:
No. This block added a slash to all dirs (that didn’t already have one) so later checks for prefixes would be on a directory boundary.

Most of the old code wasn’t very clear because it assumed cleaned paths and common prefixes. These changes explicitly clean paths and use names from file info instead of paths when filtering.